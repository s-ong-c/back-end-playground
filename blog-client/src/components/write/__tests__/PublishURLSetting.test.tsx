import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import PublishURLSetting, {
  PublishURLSettingProps,
} from '../PublishURLSetting';

describe('PublishURLSetting', () => {
  const setup = (props: Partial<PublishURLSettingProps> = {}) => {
    const initialProps: PublishURLSettingProps = {
      username: 'SONGC',
      urlSlug: 'url-slug',
      onChangeUrlSlug: () => {},
    };
    const utils = render(<PublishURLSetting {...initialProps} {...props} />);
    return {
      ...utils,
    };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('show username props', () => {
    const utils = setup();
    utils.getByText('/@SONGC/');
  });
  it('urlSlug is working property', () => {
    const onChangeUrlSlug = jest.fn();
    const utils = setup({
      onChangeUrlSlug,
    });
    const input = utils.getByValue('url-slug');
    fireEvent.change(input, { target: { value: 'hello-world' } });
    expect(onChangeUrlSlug).toBeCalledWith('hello-world');
  });
});
