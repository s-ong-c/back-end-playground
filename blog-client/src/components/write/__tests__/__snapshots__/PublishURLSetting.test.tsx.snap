// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishURLSetting matches snapshot 1`] = `
<div>
  <section
    class="sc-bdVaJa jSADnk sc-bwzfXH uTxCW"
  >
    <h3>
      URL 설정
    </h3>
    <div
      class="contents"
    >
      <div
        class="sc-htpNat aiqNY"
      >
        <div
          class="username"
        >
          /@
          SONGC
          /
        </div>
        <input
          class="sc-bxivhb knHhGg"
          value="url-slug"
        />
      </div>
    </div>
  </section>
</div>
`;
